### PR TITLE
Work towards specified API

### PR DESCRIPTION
**Pull request by michaelfig**
Ported from Agoric/eventual-send#43 to agoric-sdk.

### CODE REVIEW

**Comment by michaelfig** [comment]
This change is to work toward implementing the new `HandledPromise` API, as well as bringing a HandledPromise endowment alongside tildot expansion. 

Please review the portions that affect you.  Thanks!

**Review by katelynsills — APPROVED** [review]
The portions relating to ERTP look good to me

**Comment by michaelfig** [comment]
@warner, @erights, @jfparadis, can you please have a look at this for changes in code you're interested in?

File: packages/default-evaluate-options/package.json [hunk 1]
```diff
@@ -23,10 +24,8 @@
   },
   "dependencies": {
     "@agoric/babel-parser": "^7.6.4",
-    "@agoric/eventual-send": "^0.4.4",
-    "@agoric/transform-eventual-send": "^1.0.3",
-    "@babel/generator": "^7.6.4",
-    "esm": "^3.2.5"
+    "@agoric/transform-eventual-send": "^1.0.4",
```
Comment (warner) [inline packages/default-evaluate-options/package.json]: this asks for `transform-eventual-send` 1.0.4, but that package was upgraded from `1.0.3` to `1.1.0` . Should this be changed to require `1.1.0`? Or can `default-eventual-options` use the previous version without problems?

File: packages/agoric-cli/package.json [hunk 2]
```diff
@@ -23,9 +23,9 @@
   },
   "dependencies": {
     "@agoric/bundle-source": "^1.0.3",
-    "@agoric/captp": "^1.0.2",
+    "@agoric/captp": "^1.2.0",
     "@agoric/evaluate": "^2.1.3",
```
Comment (warner) [inline packages/agoric-cli/package.json]: The `evaluate` dependency remains at `2.1.3`, but `evaluate` was upgraded to `2.2.0` . Other packages that depend upon `evaluate` had their dependencies increased. Should this one be increased too? Or can `agoric-cli` use the older `evaluate` without problems?

**Review by warner — CHANGES_REQUESTED** [review]
The swingset/cosmic-swingset parts look ok to me.

This needs to be rebased to current trunk before landing. Zoe was split out into a separate package/workspace in that history, so you should look at the new ERTP/Zoe dependencies on the changed modules and make sure they look good (in particular I think zoe's dependencies on `eventual-send`, `marshal`, and `evaluate` may need to be updated).

File: packages/agoric-cli/package.json (hunk 2 above)
Comment (michaelfig) [reply packages/agoric-cli/package.json]: Should be increased.  I've done that now.

File: packages/default-evaluate-options/package.json (hunk 1 above)
Comment (michaelfig) [reply packages/default-evaluate-options/package.json]: It should be changed.  Thanks.

**Review by warner — APPROVED** [review]
that looks good, thanks

File: packages/SwingSet/src/controller.js [hunk 3]
```diff
@@ -63,14 +58,17 @@ function getKernelSource() {
 function makeEvaluate(e) {
```
[the remark this answers is not in the document]
Comment (michaelfig) [reply packages/SwingSet/src/controller.js:58]: We still have it for now.  Thus will change soon in a future pr.

**Pull request merged**